### PR TITLE
chore: upgrade kokoro test to use latest go version

### PR DESCRIPTION
**Pull request by aabmass**

### CODE REVIEW

**Comment by aabmass** [comment]
Kokoro tests should pass now that https://github.com/googleapis/google-cloud-go/pull/10484 is merged. Just need Go to pick up the new HEAD ref

File: system-test/integration_test.sh [hunk 1]
```diff
@@ -29,21 +29,19 @@ export GOOGLE_APPLICATION_CREDENTIALS="${SERVICE_KEY}"
...
 export PATH=$PATH:/usr/local/go/bin
+go version
 
 # Initializing go modules allows our dependencies to install versions of their
 # dependencies specified by their go.mod files. This reduces the likelihood of
 # dependencies breaking this test.
-go version
 go mod init e2e
-retry go get cloud.google.com/go/profiler/proftest@HEAD
+retry go get cloud.google.com/go/profiler/proftest@main
```
Comment (aabmass) [inline system-test/integration_test.sh:44]: For some reason `@HEAD` is pulling in the last release and not actually pulling from HEAD to pick up https://github.com/googleapis/google-cloud-go/pull/10484. This seems to fix it for now to actually pull in the latest.

Comment (psx95) [reply system-test/integration_test.sh:44]: Quick verification - Do we actually want to always fetch from latest commit instead of latest release?

**Review by psx95 — APPROVED** [review]
LGTM.

**Pull request merged**